ports/esp8266/esppwm.c

```diff
@@ -126,6 +128,11 @@ pwm_start(void) {
 
     LOCK_PWM(critical);   // enter critical
 
+    // If a toggle is pending, we reset it since we're going to change again
+    mp_uint_t irq_state = mp_hal_quiet_timing_enter();
+    do_toggle = 0;
+    mp_hal_quiet_timing_exit(irq_state);
```

Comment:
I don't think it's necessary to disable IRQs here, it should be an atomic operation to store 0 to `do_toggle`.

Reply:
You're right. I'll remove it.